### PR TITLE
Remove the config datafusion.execution.coalesce_target_batch_size and use datafusion.execution.batch_size instead

### LINKED ISSUE #4756: Remove the config `datafusion.execution.coalesce_target_batch_size`
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

There's already a config of `datafusion.execution.batch_size` to indicate the row number of a batch for processing. I don't think it's necessary to make a different batch size for the `CoalesceBatches` rule.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by yahoNanJing**
# Which issue does this PR close?

Closes #4756.

# Rationale for this change

As #4756 mentioned, there's already a config of `datafusion.execution.batch_size` to indicate the row number of a batch for processing. I don't think it's necessary to make a different batch size for the CoalesceBatches rule.

# What changes are included in this PR?

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by yahoNanJing** [comment]
Hi @alamb, @andygrove, could you help review this PR?

File: datafusion/core/src/execution/context.rs [hunk 1]
```diff
@@ -1573,19 +1544,8 @@ impl SessionState {
...
-        {
-            physical_optimizers.push(Arc::new(CoalesceBatches::new(
-                config
-                    .config_options
-                    .get_u64(OPT_COALESCE_TARGET_BATCH_SIZE)
-                    .unwrap_or_default()
-                    .try_into()
-                    .unwrap(),
-            )));
+        if config.coalesce_batches() {
```
Comment (alamb) [inline datafusion/core/src/execution/context.rs:1547]: That reads much more nicely 👍

**Review by alamb — APPROVED** [review]
I think this change makes a lot of sense. Thank you @yahoNanJing 

Maybe @Dandandan  remembers why there were two different batch sizes

**Comment by alamb** [comment]
I plan to merge this later today unless there are any more comments.

**Comment by yahoNanJing** [comment]
Since there's no opposite comments, I will merge this PR.

**Pull request merged**